### DISCUSSION TITLE
go.sum hash mismatch for a single private go dependency, others seem to work

### DISCUSSION THREAD

**Discussion by ekristen** [renovatebot/renovate#25041] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

go.sum hash security error on a single private go module dependency in the same org where other private go module dependencies exist. 

For some reason I have a project (look up run id `c3b1dd92-bb81-4e79-90df-7bc17cafa82f`) that has two private go modules hosted in the same org. 1 works, the other throws a hash mismatch. The one that works is a commit ref and that project has no tags. The other that throws an error is a commit ref but has tags on it's repository. 

Beyond that both are accessible by renovate. I've tried token encrypted overwrite, I've tried GOPRIVATE custom env vars. It seems both are accessible, but the one always throws an error on go.sum hashes not matching.

Logs seem to indicate it can find the dependency, hash and version.

Given I can't reproduce it on other private repos I have, I'm unsure building a repo repository is feasible. Could this be a bad cache? 

I am running go 1.21.

I've tried GOPRIVATE, host rules with custom tokens, as far as I can tell it's not an access problem, but I have not been able to find anything in the logs to suggest why the hash mismatch is happening.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
Do you mean that Renovate has created a PR with a updated go.sum, and that PR fails your checks because of a mismatch? If not, can you explain what you mean?

**Reply by ekristen** [reply]
Renovate is creating PRs for other dependencies (ie aws SDK) and renovate/artifact fails for each one saying hash for an unrelated private dependency does not match and go.sum can't be updated.

**Reply by rarkins** [reply]
So the error is found _by_ Renovate when it runs `go get` to update artifacts?

**Reply by ekristen** [reply]
That seems to be the case

**Reply by ekristen** [reply]
I can get you a bunch of job ids to review if that'll help. 

Basically Org A has Proj B, C, D -- all use renovate. B and C reference D with no problem. Project C is the one with the problem, if references B as well, this is where the go.sum artifact has hash mismatch.

**Reply by rarkins** [reply]
If you can paste here or email me a full URL for a log then the system is set up to let me access it. It takes me much longer to take a job ID and figure out which repo it's for

**Reply by ekristen** [reply]
Sent via email.

**Comment by ekristen** [comment, marked as answer]
The problem is that the dependency is using git-lfs for some non-go related files that get bundled into the .zip file that throws the hash off. There's an open issue for git-lfs being added to renovate. https://github.com/renovatebot/renovate/issues/6842

@rarkins I took the docker run command from the logs and simply modified the command from

```console
bash -l -c "install-tool golang 1.21.1 && go get -d -t ./..."
``` 

to 

```console
bash -l -c "install-tool git-lfs v3.4.0 && git lfs install --skip-repo && install-tool golang 1.21.1 && go get -d -t ./..."
```

and everything works as expected. The solution is renovate supporting git-lfs and installing it in the base image.
